Auto build docs

brainwavecoder9 opened the pull request:
## SUMMARY

A number of changes to improve and automate the documentation build, and a little cleanup.

## EXPLANATION

When trying to run `make` to build the README for #51 I ended up down a rabbit hole trying to get my build environment to be consistent (as did someone else in #24). Even after what I thought was a successful run, I still ended up with a slightly different format, which can make a very minor change seem quite complicated. Although there are a number of changes to this README, these are expected to be normalized through the new process so future changes are expected to be consistent. 

This modification to the documentation creation process automates the build process in an isolated environment for creating the README file.

Additionally, a few other small changes were made to cleanup the repo and related process. A summary of all changes is below.

## DESCRIPTION OF CHANGES 

`.github/workflows/generate-documentation.yml`
 - New file enables the README `make` process to run within a GitHub Action. This defines dependencies and ensures consistency for the process, means anyone modifying documentation does not have to configure a local environment, and automates the process so that the make command never needs to be run manually. This will automatically run to create and commit the README, but only after a commit is made to the docs folder.

`.gitattributes`
 - modified to use lf line endings for MD files, nice for consistency generally, and required for local build process.
 
`docs/Makefile` has a number of changes: 
 - upgraded syntax for 3.x pandoc to address deprecation [WARNING]
 - _markdown section updated to only execute python scripts. a bit cleaner than blindly attempting to execute text files
 - PYTHONPATH set to fix issue where chain_server was not available
	
`docs/0_2_999_footer.md`
 - removed an empty file that is not referenced, probably left over from changes made by Ryan Kraus in June (4477e66 & bd71c95)

`docs/0_1_personal_key.md`
 - URL link fix, which was the original issue resolved as part of: #49

`docs/3.2_docs.md` 
 - Remove reference to `make`, added additional clarifying info, and let users know that the process will happen automatically

`.static/*`
 - A number of files automatically updated as part of the build process, referenced by the README

brainwavecoder9 commented:
FYI @AmeliaYe 

Happy to have a discussion here about any questions, or additional changes/updates that you might want related to this

AmeliaYe reviewed (commented):
Other than the in-line comments everything including the README seems good to me

brainwavecoder9 commented on docs/Makefile:
```diff
@@ -36,7 +37,11 @@ _render:
 	for page in $$(cat $(TOC_FILE)); do
-		./$$page 2> /dev/null || cat $$page || exit 1;
+		if [[ $$page == *.py ]]; then
+			./$$page 2> /dev/null || cat $$page || exit 1;
```
Maybe I am misunderstanding the comment but the existing `./$$page` code has not changed:
`./$$page 2> /dev/null || cat $$page || exit 1;`

The `./$$page` command was moved inside of the `if [[ $$page == *.py ]]` conditional was added to ensure that only python files are executed. 

Previously the process had attempted to execute all files in the TOC, including .md files.

brainwavecoder9 commented:
FYI @AmeliaYe comments have been resolved.

- implemented requirements change per request 
- added clarification re: concern about change affecting functionality for script execution
